audio-worklet/bitcrusher-test.html

```diff
@@ -0,0 +1,218 @@
+<!--
+Copyright 2017 Google Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<!DOCTYPE html>
+  <head>
+    <script src="src/bitcrusher-script-processor.js"></script> 
+    <script>
+      testSampleReduction();
+      testBitcrusherInAudioContext();
+      testMultipleChannels();
+      testIllegalInput();
+
+      /**
+       * Verify the sample rate reduction component of the bitcrusher operating
+       * outside of an audio context. A bitcrushed array generated from an 
+       * arbitrary sequence of numbers should hold sequences of consecutive 
+       * equal numbers, controlled by the variable factor.
+       */
+      function testSampleReduction() {
+        let context = new OfflineAudioContext(1, 44100, 44100);
+
+        for (let reduction = 1; reduction < 10; reduction++) {
+          // BufferSize chosen arbitrarily and can be any value greater than
+          // reduction to perform the test.
+          const bufferSize = 20;
+          let referenceBuffer = new Float32Array(bufferSize);
+          let outputBuffer = new Float32Array(bufferSize);
+          let bitcrusher =
+              new Bitcrusher(context, {bitDepth: 24, reduction: reduction});
+
+          for (let i = 0; i < bufferSize; i++) {
+            referenceBuffer[i] = i;
+          }
+          bitcrusher.processBuffer_(referenceBuffer, outputBuffer, 0);
+          
+          // Verify computed values match expected values.
+          outputBuffer.map((data, index) => {
+            let expected = Math.floor(index / reduction) * reduction;
+            console.assert(
+                data === expected, 
+                'computed ' + data + ' but expected ' + expected);
+          });
+        }
+      }
+
+      /**
+       * Verify that the bitcrusher does not alter samples beyond an acceptable 
+       * margin when parameterized to avoid information reduction.
+       */
+      function testBitcrusherInAudioContext() {
+        // Sample Rate is held at a power of two to avoid sample interpolation
+        // from the delay node. It is artificially low for easy inspection of
+        // buffers.   
+        const sampleRate = 4096;
+        let context = new OfflineAudioContext(2, sampleRate, sampleRate);
+        const reduction = 1;
+        const bitDepth = 24;
+        // BufferSize should be a power of two for synchronization with
+        // the delayed oscillator.
```

Comment:
Isn't this comment wrong?  The size needs to be a power of two because you're using a ScriptProcessor.